articles/data-leakage/index.md

```diff
@@ -0,0 +1,83 @@
+
+#### Implicit Leakage
+
+Consider temporal data. This is data related to time. Time is a crucial factor. An example is time-series data. Imagine having a training set with two data points, 1 and 3. Let's have the test set with a single data point labeled 2. Let's assume that the temporal sequence of these points is 1, 2, then 3. There is a likelihood that we have introduced data leakage thanks to how we created our test and training sets. Since point 3 is in the training set, and 2 is in the test set. In this situation, we are unrealistically training our model on future information compared to the position of the test data in time. We inadvertently end up leaking information about the future into the past.
+
+### Techniques to Minimize Leakage
+
+Since we now know the possible causes of leakage, we should explore the various ways to minimize or prevent data leakage. Let's go through three effective methods.
+
+#### Normalizing Correctly Before Cross-Validation
+
+It is easy to leak data when preparing data for machine learning. As a result, overfitting occurs in training data. This also impacts the performance of the model on test data. We would be given a flattering picture of performance, but it performs poorly when the model is deployed in the real world.
+For instance, data leakage will occur in a scenario where one carries out normalization of an entire dataset, then evaluates the algorithm's performance using cross-validation. [Normalization](https://developers.google.com/machine-learning/data-prep/transform/normalization#:~:text=%20Normalization%20Techniques%20at%20a%20Glance%20%201,a%20wide%20range%20to%20a%20narrow...%20More%20) is a process that aims to improve the performance of a model by transforming features to be on a similar scale. Data normalization is carried out by dividing the data by its mean.
+I described [cross validation](/engineering-education/evaluating-ml-model-performance/) as the use of two datasets to evaluate an algorithm. Specifically, there is a training set and an independent set that is partitioned from an original dataset. These splits of the dataset are called folds, which are often equal in size. A model being evaluated is trained on all folds except one. So, a model will be trained on three and tested on the remaining fold if there are four-folds. The process is repeated until each fold is used for testing.
+From the example above, a clear mistake is the normalizing of an entire dataset. We should note that normalization needs to be carried out after splitting data into train, test, and validation sets. If normalization is carried out before splitting, it will affect the whole dataset instead of the training set. That is, the mean and standard deviation used for normalization will reflect the whole dataset. As a result, information on test and validation sets sips into the training set.
```

Comment:
It is clear that normalization affects the results. But, how does the order of events change the results? Can you elaborate on that?